stacks-common/src/address/c32.rs

```diff
@@ -69,44 +97,34 @@ fn c32_encode(input_bytes: &[u8]) -> String {
     String::from_utf8(result).unwrap()
 }
 
-fn c32_normalize(input_str: &str) -> String {
-    let norm_str: String = input_str
-        .to_uppercase()
-        .replace("O", "0")
-        .replace("L", "1")
-        .replace("I", "1");
-    norm_str
-}
-
 fn c32_decode(input_str: &str) -> Result<Vec<u8>, Error> {
     // must be ASCII
     if !input_str.is_ascii() {
         return Err(Error::InvalidCrockford32);
     }
+    c32_decode_ascii(input_str)
+}
 
+fn c32_decode_ascii(input_str: &str) -> Result<Vec<u8>, Error> {
     let mut result = vec![];
     let mut carry: u16 = 0;
     let mut carry_bits = 0; // can be up to 5
 
-    let iter_c32_digits_opts: Vec<Option<usize>> = c32_normalize(input_str)
-        .chars()
-        .rev()
-        .map(|x| C32_CHARACTERS.find(x))
```

Comment:
Refactor of this `find` from O(n) to O(1) in the hot path is one of the primary optimizations in this PR